pkg/importer/imageio-datasource.go

```diff
@@ -320,9 +546,22 @@ func getTransfer(conn ConnectionInterface, diskID string) (*ovirtsdk4.ImageTrans
 		} else if phase == ovirtsdk4.IMAGETRANSFERPHASE_TRANSFERRING {
 			break
 		} else {
+			cleanupTransfer(conn, it)
```

Comment:
maybe better as a defer? At a minimum check the returned err

Reply:
This one is actually identical to all the other cleanupTransfers in createImageioReader, but it shouldn't be moved out of getTransfer because it's not certain there's really a transfer to clean up until this point. Either way, I still don't think checking the error will add much because there's already lots of logging inside cleanupTransfer and acting on an error from cleanupTransfer might obscure the original reason for failure.

Reply:
This is kind of like checking a close return value in a defer on a file descriptor. There is not much you can do if an error occurs, but purists will argue you should check all errors. I would argue if we are not interested in the error from the cleanupTransfer, then why are we returning it? If we already log the error inside the function and don't want calling functions to do anything with it. Then there is no sense that it is a return parameter.

Reply:
I looked closer at all the cleanupTransfer calls, and I think they can be condensed in a way that it makes more sense to always check its return value. I am testing it out to make sure disks still get unlocked when they should be.